src/NuGet.Core/NuGet.Packaging/Signing/Authoring/Signer.cs

```diff
@@ -14,63 +14,80 @@ namespace NuGet.Packaging.Signing
     /// </summary>
     public sealed class Signer
     {
-        private readonly ISignedPackage _package;
         private readonly SigningSpecificationsV1 _specifications = SigningSpecifications.V1;
-        private readonly ISignatureProvider _signatureProvider;
+        private readonly SignerRequest _request;
 
         /// <summary>
         /// Creates a signer for a specific package.
         /// </summary>
         /// <param name="package">Package to sign or modify.</param>
-        public Signer(ISignedPackage package, ISignatureProvider signatureProvider)
+        public Signer(SignerRequest request)
         {
-            _package = package ?? throw new ArgumentNullException(nameof(package));
-            _signatureProvider = signatureProvider ?? throw new ArgumentNullException(nameof(signatureProvider));
+            _request = request ?? throw new ArgumentNullException(nameof(request));
```

Comment:
use Assumes.Present from the Microsoft namespace.

Reply:
`Assumes.Present(...)` comes from the [Microsoft.VisualStudio.Validation](https://www.nuget.org/packages/Microsoft.VisualStudio.Validation/) package, which shouldn't be pulled in by NuGet.Packaging.